File: 4/task.ex
defmodule Task do

  def min2(count, values) do
    min2(count, values, [])
  end

  defp min2(_, [], result) do
    Enum.reverse(result)
  end

  defp min2(count, [head_value|tail_values], result) do
    {value1, value2} = head_value
    value3 = case value1 < value2 do
      true -> value1
      false -> value2
    end
    min2(count - 1, tail_values, [value3|result])
  end

end

File: 17-triangles/task.erl
-module(task).

-export([is_triangle/2]).


is_triangle(Count, Values) ->
  is_triangle2(Count, Values, []).

is_triangle2(_, [], Result) ->
  lists:reverse(Result);
is_triangle2(Count, [{V1, V2, V3}|ValuesTail], Result) when V1 >= V2 + V3; V2 >= V1 + V3; V3 >= V1 + V2 ->
  is_triangle2(Count - 1, ValuesTail, [0|Result]);
is_triangle2(Count, [{V1, V2, V3}|ValuesTail], Result) ->
  is_triangle2(Count - 1, ValuesTail, [1|Result]).

